### PR TITLE
Color schema

### PR DESCRIPTION
**Pull request by euri10**
## Change Summary

Added schema generation capability for the Color type.

I'm not too sure about the correctness of the generated schema @tiangolo but I checked http://json-schema.org/latest/json-schema-validation.html and see no mention of anything specific to colors appart at the very bottom where it says:

> "format": remove attributes "phone", "style", "color"; rename "ip-address" to "ipv4"; add references for all attributes.

Tests don't pass 3.8-dev

## Related issue number

https://github.com/tiangolo/fastapi/issues/354

## Checklist

* [x ] Unit tests for the changes exist
* [ ] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `HISTORY.rst` has been updated
  * if this is the first change since a release, please add a new section
  * include the issue number or this pull request number `#<number>`
  * include your github username `@<whomever>`

### CODE REVIEW

**Comment by samuelcolvin** [comment]
@euri10 it looks like you've broken docs build, please could you review.

**Comment by euri10** [comment]
> @euri10 it looks like you've broken docs build, please could you review.

sorry for that, should be ok

**Comment by samuelcolvin** [comment]
no problem, easy to do. 

@tiangolo please let us know if this is correct from the point of view of JSON schema.

Ignore the coverage drop, I think it's fictional.

File: pydantic/schema.py [hunk 1]
```diff
@@ -679,6 +680,7 @@ def field_singleton_sub_fields_schema(
     (list, {'type': 'array', 'items': {}}),
     (tuple, {'type': 'array', 'items': {}}),
     (set, {'type': 'array', 'items': {}, 'uniqueItems': True}),
+    (Color, {'type': 'string'}),
```
Comment (tiangolo) [inline pydantic/schema.py]:
```suggestion
    (Color, {'type': 'string', 'format': 'color'}),
```

File: docs/schema_mapping.py [hunk 2]
```diff
@@ -433,7 +433,14 @@
         '',
         'JSON Schema Core',
         'All the properties defined will be defined with standard JSON Schema, including submodels.'
-    ]
+    ],
+    [
+        'Color',
+        'string',
+        '',
```
Comment (tiangolo) [inline docs/schema_mapping.py]:
```suggestion
        '{"format": "color"}',
```

File: tests/test_schema.py [hunk 3]
```diff
@@ -1330,3 +1331,16 @@ class Model(BaseModel):
...
+
+def test_color_type():
+    class Model(BaseModel):
+        color: Color
+
+    model_schema = Model.schema()
+    assert model_schema == {
+        'title': 'Model',
+        'type': 'object',
+        'properties': {'color': {'title': 'Color', 'type': 'string'}},
```
Comment (tiangolo) [inline tests/test_schema.py]:
```suggestion
        'properties': {'color': {'title': 'Color', 'type': 'string', 'format': 'color'}},
```

**Comment by tiangolo** [comment]
Thanks for your patience guys. Pretty hectic last days.

**Comment by samuelcolvin** [comment]
looks like a few more things to fix, also please resolve conflicts.

**Comment by euri10** [comment]
> looks like a few more things to fix, also please resolve conflicts.

looks like I fucked up the rebase

**Comment by samuelcolvin** [comment]
> > looks like a few more things to fix, also please resolve conflicts.
> 
> looks like I fucked up the rebase

Don't worry, these things happen. New PR?

**Comment by euri10** [comment]
> > > looks like a few more things to fix, also please resolve conflicts.
> > 
> > 
> > looks like I fucked up the rebase
> 
> Don't worry, these things happen. New PR?

yep sorry about that, new one on https://github.com/samuelcolvin/pydantic/pull/661

**Pull request closed without merging**